### PR TITLE
Adding piece of code to verify binary sign

### PR DESCRIPTION
**Pull request by robsonala**

### CODE REVIEW

File: src/CryptoHelper.php [hunk 1]
```diff
@@ -112,7 +135,20 @@ public static function verify($data, $caInfo = null, $rootCerts = null)
 
         $outFile = self::getTempFilename();
 
-        return openssl_pkcs7_verify($data, $flags, $outFile, $rootCerts) === true;
+        $out = openssl_pkcs7_verify($data, $flags, $outFile, $rootCerts) === true;
+
+        if ($out === false && $encodeData === false) {
+            $originalCall = func_get_args();
```
Comment (enforcebeautifulcode) [inline src/CryptoHelper.php]: Why do you need to use `func_get_args`?

Just use the args directly `$data, $caInfo = null, $rootCerts = null`

File: src/CryptoHelper.php [hunk 2]
```diff
@@ -86,14 +95,28 @@ public static function sign($data, $cert, $privateKey = null, $headers = [], $mi
...
+            $temp = new MimePart($data->getHeaders());
+            foreach ($data->getParts() as $part) {
+                if (self::stringIsBinary($part->getBody())) {
+                    $encodedPart = new MimePart($part->getHeaders(), Utils::encodeBase64($part->getBody()));
+                    $temp->addPart($encodedPart);
+                } else {
+                    $temp->addPart($part);
+                }
+            }
+            $data = $temp;
```
Comment (enforcebeautifulcode) [inline src/CryptoHelper.php]: Wouldn't it be better to extract this to an "encodePayload" method?

File: src/CryptoHelper.php [hunk 3]
```diff
@@ -8,6 +8,15 @@
  */
 class CryptoHelper
 {
+    protected static function stringIsBinary($str): bool
+    {
+        $str = str_ireplace("\t", "", $str);
+        $str = str_ireplace("\n", "", $str);
+        $str = str_ireplace("\r", "", $str);
```
Comment (enforcebeautifulcode) [inline src/CryptoHelper.php]: You could just do `str_ireplace(["\t", "\n", "\r"], "", $str)`

File: src/CryptoHelper.php (hunk 1 above)
Comment (robsonala) [reply src/CryptoHelper.php]: $data and $rootCerts is modified, or I use func_get_args or I need to put it in a variable before the update

File: src/CryptoHelper.php (hunk 3 above)
Comment (robsonala) [reply src/CryptoHelper.php]: 👍

File: src/CryptoHelper.php (hunk 1 above)
Comment (enforcebeautifulcode) [reply src/CryptoHelper.php]: Leave as it. Makes sense :)

**Review by grahamtilson — APPROVED** [review]

File: src/CryptoHelper.php (hunk 2 above)
Comment (robsonala) [reply src/CryptoHelper.php]: 👍

**Review by enforcebeautifulcode — APPROVED** [review]

**Pull request merged**